### PR TITLE
CI: Fix test-runner build step

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

Chromatic was failing on play function errors and test-runner was swallowing them because of a bug in the test-runner https://github.com/storybookjs/test-runner/issues/195

## What I did

Upgrade test-runner

## How to test

- [ ] CI fails consistently in #19256

### CODE REVIEW

File: scripts/sandbox.ts [hunk 1]
```diff
@@ -310,10 +310,18 @@ async function updateStoriesField(mainConfig: ConfigFile, isJs: boolean) {
...
     ? stories.map((specifier) => specifier.replace('js|jsx|ts|tsx', 'js|jsx'))
     : stories;
 
+  if (process.env.CI) {
+    logger.log('Skipping error stories to avoid failures in CI...');
+    // TODO: once we have story tags in place, we can filter error tags instead of doing this
+    updatedStories = updatedStories.map((specifier) =>
+      specifier.replace('src/**/*.stories.@', 'src/**/!(errors)*.stories.@')
+    );
+  }
```
Comment (tmeasday) [inline scripts/sandbox.ts:323]: @shilman don't all our tools have parameters that allow us to skip certain stories?

Comment (tmeasday) [reply scripts/sandbox.ts:323]: @yannbf

**Comment by shilman** [comment]
@tmeasday @IanVS @yannbf this PR upgrades the test runner so that it actually tests (earlier versions were swallowing 7.0 errors). However, CI is currently breaking with "Cannot call renderSelection before initialization" in Vite projects, which was fixed here https://github.com/storybookjs/storybook/pull/19071. Any idea why it might be failing in the test runner?

File: code/renderers/react/template/stories/errors.stories.tsx [hunk 2]
```diff
@@ -9,6 +9,10 @@ export default {
     storyshots: { disable: true },
     chromatic: { disable: true },
   },
+  decorators: [
+    // Skip errors if we are running in the test runner
+    (storyFn: any) => window?.navigator?.userAgent?.match(/StorybookTestRunner/) || storyFn(),
```
Comment (tmeasday) [inline code/renderers/react/template/stories/errors.stories.tsx:14]: Should we make this a project-level decorator that looks at a parameter?

**Review by tmeasday — APPROVED** [review]
LGTM apart from question about parameter.

**Comment by tmeasday** [comment]
>  Any idea why it might be failing in the test runner?

🤔 hmmm, that issue is due to timing related stuff so I guess it could just be a new kind of timing triggering a similar kind of issue. Not sure.

File: code/renderers/react/template/stories/errors.stories.tsx (hunk 2 above)
Comment (tmeasday) [reply code/renderers/react/template/stories/errors.stories.tsx:14]: There's no simple way to do this so let's skip it for now.

**Pull request merged**